File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> ans = new ArrayList<>();
        int[] toAdd = newInterval;
        for(int i =0;i<intervals.length;i++){
            //end < start
            if(intervals[i][1] < toAdd[0]){
                ans.add(intervals[i]);
                //start > end
            }else if(intervals[i][0] > toAdd[1]){
                ans.add(toAdd);
                toAdd = intervals[i];
                //if there is an overlapping
            }else{
                toAdd = new int[]{Math.min(intervals[i][0],toAdd[0]),Math.max(intervals[i][1],toAdd[1])};
            }
        }
       
            ans.add(toAdd);
        return ans.toArray(new int[ans.size()][2]);
    }
}

File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        int n = intervals.size();
        vector<vector<int>> ans;
        
        for(int i = 0; i < n; i++) {
            if(intervals[i][0] > newInterval[1]) {
                ans.push_back(newInterval);
                for(int j = i;j < n; j++) {
                    ans.push_back(intervals[j]);
                }
                return ans;
            }
            else if(newInterval[0] > intervals[i][1]) {
                ans.push_back(intervals[i]);
            } else {
                newInterval[0] = min(intervals[i][0],newInterval[0]);
                newInterval[1] = max(intervals[i][1],newInterval[1]);
            }
        }
        ans.push_back(newInterval);
        return ans;
    }
};